### DISCUSSION TITLE
How can i remove this payment dependency?

### DISCUSSION THREAD

**Discussion by marcheygroup** [graphcommerce-org/graphcommerce#1649] [General] [answered]
[SUCCESS] Generate ../../node_modules/@graphcommerce/graphql/generated/fragments.json
--
19:37:18.510 | [STARTED] Generate to ../../ (using EXPERIMENTAL preset "@graphcommerce/graphql-codegen-near-operation-file")
19:37:18.510 | [STARTED] Load GraphQL schemas
19:37:18.510 | [SUCCESS] Load GraphQL schemas
19:37:18.510 | [STARTED] Load GraphQL documents
19:37:18.712 | [SUCCESS] Load GraphQL documents
19:37:18.712 | [STARTED] Generate
19:37:22.494 | [FAILED] Unable to find field "createKlarnaPaymentsSession" on type "Mutation"!
19:37:22.494 | [SUCCESS] Generate outputs
19:37:22.524 | error Command failed with exit code 1.
19:37:22.524 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
19:37:22.534 | Error: Command "yarn run build" exited with 1

**Comment by marcheygroup** [comment]
@paales

**Comment by paales** [comment, marked as answer]
I would suggest not cloning the graphcommerce repo but follow the installation instructions; https://www.graphcommerce.org/docs/getting-started/create

**Reply by marcheygroup** [reply]
I was able to deploy this on vercel on last versions not sure but that is 3-4 month ago but now there are so many changes that nothing is working.

i followed your every single doc but not sure how to remove klarna payment method as per previous error.
i removed scripts and renamed scripts_local to scripts in main project package.json and now it failed earlier.
i think this is not for production but for local setup..very confused with this how its not running for my fork.
i have connected previously my instance successfully but this time nothing worked. and this Klana payment is the faulty one.

error /vercel/path0/node_modules/@graphcommerce/magento-graphcms: Command failed.
--
10:17:55.671 | Exit code: 1
10:17:55.671 | Command: patch-package
10:17:55.671 | Arguments:
10:17:55.671 | Directory: /vercel/path0/node_modules/@graphcommerce/magento-graphcms
10:17:55.671 | Output:
10:17:55.671 | patch-package 6.4.7
10:17:55.671 | Applying patches...
10:17:55.671 | Error: Patch file found for package next which is not present at node_modules/next
10:17:55.671 | ---
10:17:55.671 | patch-package finished with 1 error(s).
10:17:55.672 | info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
10:17:55.717 | Error: Command "yarn install" exited with 1

**Reply by marcheygroup** [reply]
@paales

**Reply by marcheygroup** [reply]
can anyone help me to make this up on my instance on vercel?

**Reply by paales** [reply]
@marcheygroup I think you should start from the beginning, not cloning the graphcommerce repository, but installing it through yarn. 

We don't support cloning the repo yet and connecting it to a random backend. We're working on this.

**Reply by marcheygroup** [reply]
If you dont support any random backend then this is useless for all magento users.
Also , always try to provide a solution not a bot reply.

**Reply by marcheygroup** [reply]
@paales it is not acceptable answer ,  i can go for another magento recommended solutions which will support latest magento releases
.

**Reply by paales** [reply]
@marcheygroup 

> If you don't support any random backend then this is useless for all magento users.

I think we are misunderstanding each other and perhaps I communicated a bit too quickly. **GraphCommerce _is_ compatible with any random Magento backend.**

If you follow the installation instructions [mentioned here](https://www.graphcommerce.org/docs/getting-started/create#download-the-example), everything should work. **Klarna will not be installed and doesn't need to be available in your backend.**

Because you were getting the Klarna related error, I am assuming that you've cloned the GraphCommerce repository instead of following the installation instructions. Running GraphCommerce in _monorepo mode_ works, but only works with the default available backend. This shouldn't be a concern for you as you do not need to run GraphCommerce in _monorepo mode_.

Does that help?

**Reply by paales** [reply]
As for getting it to work with any backend while running in _monorepo mode_, we've got a PR here that should fix that part: https://github.com/graphcommerce-org/graphcommerce/pull/1613, however that one isn't ready yet.

**Reply by marcheygroup** [reply]
@paales 

I know its open source but we are using latest magento release 2.4.5 and its broken at Unable to find field "createKlarnaPaymentsSession" on type "Mutation .
Your still using old klarna graphql i think and latest magento does not have this old module-kp which exists in old releases like 2.4.3.
My question is that is there any chances when it will support latest 2.4.5 release.

Other question:

**_Because you were getting the Klarna related error, I am assuming that you've cloned the GraphCommerce repository instead of following the installation instructions. Running GraphCommerce in monorepo mode works, but only works with the default available backend. This shouldn't be a concern for you as you do not need to run GraphCommerce in monorepo mode_** __

Current Situation:

1.i am deploying on vercel with passing a root path to graphcommerce/examples/magento-graphcms/
adding env variables same in your env but changed MAGENTO_ENDPOINT and NEXT_PUBLIC_GRAPHQL_ENDPOINT .

2. Deploying everything in production not in dev mode.

3. I dont know why it is picking klarna instead i did the root directory to graphcommerce/examples/magento-graphcms/

**Reply by paales** [reply]
Can you help me debug something?

Can you create a new/separate installation with these step and see if that does work?

1. `git clone https://github.com/graphcommerce-org/graphcommerce.git` — clone repository
2. `mkdir my-project` — create project folder
3. `cp -R graphcommerce/examples/magento-graphcms/. my-project && rm -rf graphcommerce` — copy example, delete repo
4. `cd my-project` — change directory to project folder
5. `cp -R .env.example .env` — create .env file
6. `rm CHANGELOG.md` — remove changelog
